viam-cartographer/src/carto_facade/carto_facade.cc

```diff
@@ -689,16 +729,30 @@ void CartoFacade::SaveInternalStateOnInterval() {
 
         std::lock_guard<std::mutex> lk(map_builder_mutex);
         map_builder.SaveMapToFile(true, filename_with_timestamp);
-        if (!started) {
+        if (state != CartoFacadeState::STARTED) {
             LOG(INFO) << "Finished saving final optimized internal state";
             break;
         }
     }
 }
 
-void CartoFacade::Stop() { StopSaveInternalState(); };
+void CartoFacade::Stop() {
+    if (state != CartoFacadeState::STARTED) {
+        LOG(ERROR) << "carto facade is in state: " << state << " expected "
+                   << CartoFacadeState::STARTED;
+        throw VIAM_CARTO_NOT_IN_STARTED_STATE;
+    }
+    state = CartoFacadeState::IO_INITIALIZED;
```

Comment:
Should we set it back to INITIALIZED?

Reply:
That makes sense, since we're not destroying CartoFacade here, just stopping the background thread that gets started in Start. Thanks for explaining!